### ISSUE TITLE
Feature Implementation Request: WaitForFirstConsumer 

### ISSUE THREAD

**Issue by ChrisPiece** [NetApp/trident#405]
Our customer thinks that everyone would benefit from this feature [WaitForFirstConsumer](https://kubernetes.io/docs/concepts/storage/storage-classes/#volume-binding-mode) that help in having the compute aligned with the storage. This feature would be useful for almost anything that pulls from a DB (mongo, cassandra, elastic ...), which are generally managed by `StateFulSets`. 

Currently, `StateFulSets` allows us to share computation across distributed AZs, however its PVCs always fall on the same AZ. Therefore, if that particular AZ needs maintenance or experiences an issue, the application would go down.

With `WaitForFirstConsumer`, the `StorageClassName` would be assigned based on the topology, selectors, anti-affinity policies, etc. etc. and would allow a more flexible management.
This feature is available on AWS, GCP and Azure that heavily use multi-AZs, but it would be useful for our customer environment too (and many others).

**Comment by gnarl** [comment]
Does your customer have multiple on-prem NetApp backends in different AZs? 

Also, from the document you linked, the `volumeBindingMode` is a field in the `StorageClass` configuration. It doesn't appear that the`StorageClassName` can be assigned by Trident based on topology.

**Comment by ChrisPiece** [comment]
Customer is trying to move to a Multi-AZ infrastructure with multiple NetApp backends, correct.
That document lists some plugins that support `WaitForFirstConsumer` and they would like to implement that functionality.
Currently they use code like this:
```
    storageSpec:
      volumeClaimTemplate:
        spec:
          storageClassName: sc-dev-az1           <----- here goes the AZ, it is hardcoded.
          accessModes: ["ReadWriteMany"]
          resources:
            requests:
              storage: 40Gi

```
The goal would be to have a dynamic `storageClassName` option so that if that AZ fails, automatically the storage class could automatically switch.
They are using OpenShift 4.3 and wonder if this functionality could be integrated with Trident.

**Comment by gnarl** [comment]
@ChrisPiece thanks for the additional information!

**Comment by adkerr** [comment]
Added in df73a8d1588300380d87eaa95c7c955aba0974a5

**Comment by gnarl** [comment]
This feature will be available in the Trident 20.10.0 release.
